codespan-lsp/src/lib.rs

```diff
@@ -229,8 +229,9 @@ where
     let source = source.as_ref();
 
     let line_span = files.line_range(file_id, position.line as usize).unwrap();
+    let line_str = source.get(line_span.clone()).unwrap();
 
-    let byte_offset = character_to_line_offset(source, position.character)?;
+    let byte_offset = character_to_line_offset(line_str, position.character)?;
```

Comment:
What was the original issue with this?

Reply:
Ohhh, it was getting the character offset in the whole file, then adding it to the line start position.